pinot-core/src/main/java/com/linkedin/pinot/core/minion/SegmentPurger.java

```diff
@@ -66,29 +89,29 @@ public File purgeSegment() throws Exception {
     String segmentName = segmentMetadata.getName();
     LOGGER.info("Start purging table: {}, segment: {}", tableName, segmentName);
 
-    try (PurgeRecordReader purgeRecordReader = new PurgeRecordReader()) {
-      SegmentGeneratorConfig config = new SegmentGeneratorConfig(purgeRecordReader.getSchema());
-      config.setOutDir(_workingDir.getPath());
-      config.setTableName(tableName);
-      config.setSegmentName(segmentName);
-      // Keep index creation time the same as original segment because both segments use the same raw data.
-      // This way, for REFRESH case, when new segment gets pushed to controller, we can use index creation time to
-      // identify if the new pushed segment has newer data than the existing one.
-      config.setCreationTime(String.valueOf(segmentMetadata.getIndexCreationTime()));
-
-      StarTreeMetadata starTreeMetadata = segmentMetadata.getStarTreeMetadata();
-      if (starTreeMetadata != null) {
-        config.enableStarTreeIndex(StarTreeIndexSpec.fromStarTreeMetadata(starTreeMetadata));
-      }
+    SegmentGeneratorConfig config = new SegmentGeneratorConfig(_purgeRecordReader.getSchema());
```

Comment:
Don't remove this try-with-resource block because it will close the record reader

Reply:
My concern with returning null was making sure is intentional - for instance, if we somehow return null for a different reason, I would want to know/decouple that. Thoughts?

Reply:
IMO, we should only return purged file, nulll, or throw exception. So using null to mark nothing need to be purged should be fine.

Reply:
makes sense, changed